src/implementations/vanilla/components/modal/modal.js

```diff
@@ -0,0 +1,198 @@
+import { queryAll } from '@ecl/dom-utils';
+import { createFocusTrap } from 'focus-trap';
+
+/**
+ * @param {HTMLElement} element DOM element for component instantiation and scope
+ * @param {Object} options
+ * @param {String} options.closeSelector Selector for closing the modal
+ * @param {Boolean} options.attachClickListener Whether or not to bind click events on toggle
+ * @param {Boolean} options.attachKeyListener Whether or not to bind keyboard events
+ */
+export class Modal {
+  /**
+   * @static
+   * Shorthand for instance creation and initialisation.
+   *
+   * @param {HTMLElement} root DOM element for component instantiation and scope
+   *
+   * @return {Modal} An instance of Modal.
+   */
+  static autoInit(root, { MODAL: defaultOptions = {} } = {}) {
+    const modal = new Modal(root, defaultOptions);
+    modal.init();
+    root.ECLModal = modal;
+    return modal;
+  }
+
+  constructor(
+    element,
```

Comment:
it feels a bit odd, our "modal" element is actually a trigger button that only takes care of opening our modal if it finds that with a query on the dom.
This means that the only possible way to initalise a modal is by having an element that identifies the modal via an aria-controls attribute.
I think this should be one of the ways to launch a modal, by calling a method in its class that shows the Element of the class (this) and this should be also possible without clicking anywhere.
This is probably why this script is so similar to the popover one, even though the two components are different: it makes sense to "attach" a popover to an element that will reveal it, since it's a sort of a tooltip, indeed, but a modal is something different...

Reply:
I'm not sure I follow you here. For the modal, in any case, we have to have a trigger (button, link or whatever), saying "when you click on me it will open this modal". That's the puropse of the aria-controls.

Reply:
yep, the component is a modal, so a dialog on top of a backdrop thingy . You could potentially want to show it without any previous interaction by the user or element controlling it the same way you could have multiple triggers for the same modal.
In any case, we can ship it like this and see the reactions, it is true that in most cases the way to interact with a modal would be a single element in the page.

Reply:
i would go for that, it seems the right way